packages/apputils-extension/src/statusbarplugin.ts

```diff
@@ -26,26 +26,26 @@ import { Title, Widget } from '@lumino/widgets';
 export const kernelStatus: JupyterFrontEndPlugin<IKernelStatusModel> = {
   id: '@jupyterlab/apputils-extension:kernel-status',
   autoStart: true,
-  requires: [IStatusBar, ITranslator],
+  requires: [IStatusBar],
   provides: IKernelStatusModel,
-  optional: [ISessionContextDialogs, ILabShell],
+  optional: [ISessionContextDialogs, ITranslator, ILabShell],
   activate: (
     app: JupyterFrontEnd,
     statusBar: IStatusBar,
-    translator: ITranslator,
     sessionDialogs: ISessionContextDialogs | null,
+    translator: ITranslator | null,
     labShell: ILabShell | null
   ): IKernelStatusModel => {
+    translator = translator ?? nullTranslator;
+    sessionDialogs =
+      sessionDialogs ?? new SessionContextDialogs({ translator });
     // When the status item is clicked, launch the kernel
     // selection dialog for the current session.
     const changeKernel = async () => {
       if (!item.model.sessionContext) {
         return;
       }
-      await (sessionDialogs ?? sessionContextDialogs).selectKernel(
-        item.model.sessionContext,
-        translator
-      );
+      await sessionDialogs!.selectKernel(item.model.sessionContext);
```

Comment:
Similar to https://github.com/jupyterlab/jupyterlab/pull/12858#pullrequestreview-1330378877 we can probably avoid the `!` here.